src/dynamic_fmt.c

```diff
@@ -380,12 +383,13 @@ char *RemoveHEX(char *output, char *input)
 	*cpo++ = *cpi;
 	cpi += 5;
 	while (*cpi) {
-		if (*cpi == '0' && cpi[1] == '0') {
-			strcpy(output, input);
-			return output;
-		}
 		if (atoi16[ARCH_INDEX(*cpi)] != 0x7f && atoi16[ARCH_INDEX(cpi[1])] != 0x7f) {
-			*cpo++ = atoi16[ARCH_INDEX(*cpi)]*16 + atoi16[ARCH_INDEX(cpi[1])];
+			char c = atoi16[ARCH_INDEX(*cpi)]*16 + atoi16[ARCH_INDEX(cpi[1])];
+			if ((only_null && c == 0) || (!only_null && (c < ' ' || c > 0x7e))) {
+				strcpy(output, input);
+				return output;
+			}
+			*cpo++ = c;
 			cpi += 2;
```

Comment:
And here. Maybe add a comment on the assumption this function makes on its input and output sizes.